src/iterators.jl

```diff
@@ -156,6 +159,47 @@ end
     reinit!(fv, fc.cc, fc.current_faceid[])
 end
 
+"""
+    InterfaceCache(grid::Grid)
+    InterfaceCache(dh::AbstractDofHandler)
+Create a cache object with pre-allocated memory for the coordinates and faces of an
+interface. The cache is updated for a new cell by calling `reinit!(cache, face_a, face_b)` where
+`face_a::FaceIndex` and `face_b::FaceIndex` are the interface faces.
+**Struct fields of `InterfaceCache`**
+ - `ic.face_a :: FaceCache`: current cell node coordinates
+ - `ic.face_b :: FaceCache`: neighbor cell node coordinates
+**Methods with `InterfaceCache`**
+ - `reinit!(cache::InterfaceCache, face_a::FaceIndex, face_b::FaceIndex)`: reinitialize [`InterfaceCache`](@ref)
+ - `cellid(ic)`: get cell ids of the current interface
+ - `getnodes(ic, use_cell_a = true)`: get the global node ids of cell A or cell B of the interface
+ - `get_cell_coordinates(ic, use_cell_a = true)`: get the coordinates of cell A or cell B of the interface
+ - `interfacedofs(ic)`: get the global dof ids of the interface cells
+ - `interfacedofranges(ic)`: get the interface dof ranges of the interface cells
+See also [`InterfaceIterator`](@ref).
+"""
+struct InterfaceCache{FC<:FaceCache}
+    face_a::FC
+    face_b::FC
+end
+
+function InterfaceCache(gridordh::Union{AbstractGrid, AbstractDofHandler})
+    face_a = FaceCache(gridordh)
+    face_b = FaceCache(gridordh)
+    return InterfaceCache(face_a, face_b)
+end
+
+function reinit!(cache::InterfaceCache, face_a::FaceIndex, face_b::FaceIndex)
+    reinit!(cache.face_a, face_a)
+    reinit!(cache.face_b, face_b)
+    return cache
+end
+
+getnodes(ic::InterfaceCache, use_cell_a::Bool = true) = getnodes(use_cell_a ? ic.face_a : ic.face_b)
+get_cell_coordinates(ic::InterfaceCache, use_cell_a::Bool = true) = get_cell_coordinates(use_cell_a ? ic.face_a : ic.face_b)
+cellid(ic::InterfaceCache) = (cellid(ic.face_a), cellid(ic.face_b))
+interfacedofs(ic::InterfaceCache) = vcat(celldofs(ic.face_a), celldofs(ic.face_b))
+interfacedofranges(ic::InterfaceCache) = (1 : length(celldofs(ic.face_a)), length(celldofs(ic.face_a)) + 1 : length(celldofs(ic.face_a))  + length(celldofs(ic.face_b)))
```

Comment:
What is the indended use for this?

Reply:
Can be used for 
```julia
function_(value|gradient|...)(iv::InterfaceValues, qp::Int, ue::Vector{Float64}, dof_range_a::UnitRange, dof_range_b::UnitRange)
```
?

Reply:
Hmm, but the dof ranges there would be dof ranges for the fields within each "interface dofrange", i.e. to select the u-dofs if you have both u and p.